polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/PageToken.java

```diff
@@ -51,13 +54,25 @@ public static PageToken fromLimit(Integer pageSize) {
   public static PageToken build(String token, Integer pageSize) {
     if (token == null || token.isEmpty()) {
       if (pageSize != null) {
-        return new LimitPageToken(pageSize);
+        return new EntityIdPageToken(pageSize);
       } else {
         return new ReadEverythingPageToken();
       }
     } else {
-      // TODO implement, split out by the token's prefix
-      throw new IllegalArgumentException("Unrecognized page token: " + token);
+      try {
+        String[] parts = token.split("/");
+        if (parts.length < 1) {
+          throw new IllegalArgumentException("Invalid token format: " + token);
+        } else if (parts[0].equals(EntityIdPageToken.PREFIX)) {
+          int resolvedPageSize = pageSize == null ? Integer.parseInt(parts[2]) : pageSize;
+          return new EntityIdPageToken(Long.parseLong(parts[1]), resolvedPageSize);
+        } else {
+          throw new IllegalArgumentException("Unrecognized page token: " + token);
+        }
+      } catch (NumberFormatException | IndexOutOfBoundsException e) {
+        LOGGER.debug(e.getMessage());
+        throw new IllegalArgumentException("Invalid token format: " + token);
+      }
```

Comment:
I, personally, find this fragment a bit more complex than it may need to be. Is there a reason why we cannot defensively check for the right amount array length after splitting it right away? Same for the NumberFormationException?

Reply:
I wanted to structure the code in a way that obviously leaves the door open for other `PageToken` implementations -- those would have different array length expectations. So we check the prefix first, and then parse the token using the logic appropriate for the `PageToken` implementation that the prefix corresponds to. 

In the old PR, there were 2 parseable `PageToken` implementations. I do agree that it looks a little clunky with the single `PageToken` implementation we have now. If this really is too confusing I can simplify this logic and then we can re-complicate it if/when we add a new `PageToken`.